CHANGELOG.md

```diff
@@ -1,3 +1,528 @@
+  ([#409](https://github.com/kinvolk/lokomotive/pull/409)).
+- Refactor Terraform executor ([#794](https://github.com/kinvolk/lokomotive/pull/794)).
+- Pass kubeconfig content around rather than a file path
+  ([#631](https://github.com/kinvolk/lokomotive/pull/631)).
+
+### Upgrading from v0.3.0
+
+#### Lokoctl Host binary upgrades
+
+##### terraform-provider-ct
+
+- Update the `ct` Terraform provider to `v0.6.1`, find the install instructions
+  [here](https://github.com/poseidon/terraform-provider-ct/blob/v0.6.1/README.md#install).
+
+#### Disable TLS Bootstrap
+
+In this release we introduced TLS bootstrapping and we enable it by default. To avoid cluster
+recreation, disable it by adding the following attribute to the `cluster ...` block:
+
+```tf
+cluster "packet" {
+  enable_tls_bootstrap = false
+...
+```
+
+#### Cluster upgrade steps
+
+Go to your cluster's directory and run the following command:
+
+```
+lokoctl cluster apply --skip-components -v
+```
+
+The update process typically takes about 10 minutes.
+After the update, running `lokoctl health` should result in an output similar to the following.
+
+```
+Node                     Ready    Reason          Message
+
+lokomotive-controller-0  True     KubeletReady    kubelet is posting ready status
+lokomotive-1-worker-0    True     KubeletReady    kubelet is posting ready status
+lokomotive-1-worker-1    True     KubeletReady    kubelet is posting ready status
+lokomotive-1-worker-2    True     KubeletReady    kubelet is posting ready status
+Name      Status    Message              Error
+
+etcd-0    True      {"health":"true"}
+```
+
+#### Cluster nodes component upgrade (optional)
+
+- Manually upgrade etcd following the steps mentioned in the doc
+  [here](https://github.com/kinvolk/lokomotive/blob/v0.4.0/docs/how-to-guides/upgrade-etcd.md).
+- Manually upgrade the kubelet running on the nodes, by following the steps mentioned in the doc
+  [here](https://github.com/kinvolk/lokomotive/blob/v0.4.0/docs/how-to-guides/upgrade-bootstrap-kubelet.md).
+
+#### Manual Cluster Changes
+
+The latest version of Metallb changes the labels of the ingress nodes. Label all the nodes that have
+`asn` set with the new labels:
+
+```
+kubectl label $(kubectl get nodes -o name -l metallb.universe.tf/my-asn) \
+  metallb.lokomotive.io/my-asn=65000 metallb.lokomotive.io/peer-asn=65530
+```
+
+Find a peer address of a node and assign it new label:
+
+```bash
+for node in $(kubectl get nodes -o name -l metallb.universe.tf/peer-address); do
+  peer_ip=$(kubectl get $node -o jsonpath='{.metadata.labels.metallb\.universe\.tf/peer-address}')
+  kubectl label $node metallb.lokomotive.io/peer-address=$peer_ip
+done
+```
+
+Now it is safe to update:
+
+```
+lokoctl component apply metallb
+```
+
+#### Ceph Upgrade steps
+
+These steps are curated from the upgrade doc provided by rook:
+https://rook.io/docs/rook/master/ceph-upgrade.html.
+
+- Keep note of the CSI images:
+
+```
```

Comment:
Sorry for being pedantic, but can you update the Ceph upgrade steps so they are correctly indented under the bullet points like we did for the UX section? :pray: